projects/self-creation-portal/src/app/services/common-service/common.service.ts

```diff
@@ -53,7 +53,23 @@ export class CommonService {
 
   clearQueryParams() {
     const queryParams = this.route.snapshot.queryParams;
-    if (Object.keys(queryParams).length > 0) {
+    const currentPath = this.router.url.split('?')[0];
+    const parentRoutes = this.route.parent?.snapshot.url.map(segment => segment.path).join('/') || '';
+    const childRoutes = this.route.routeConfig?.children?.map(route => `${parentRoutes}/${route.path}`) || [];
+    
+    // Filter relevant sidenav paths that match the routes
+    const sidenavPaths = childRoutes.filter(routePath => 
+      ['drafts', 'up-for-review', 'submitted-for-review'].some(sidenav => routePath.includes(sidenav))
```

Comment:
As I already mentioned don't use string comparison of project status, use constant variables